core/src/error/mod.rs

```diff
@@ -70,6 +70,21 @@ impl Error {
         Error::duplicate_field(&path_to_string(path))
     }
 
+    /// Creates a new error for an attribute that is a single name-value pair, as opposed to the
+    /// expected `Meta::List` form.
+    pub fn name_value_attribute() -> Self {
```

Comment:
This has been solved by moving the parsing of attributes into meta-lists inside `darling::util`, so that better diagnostics do not require future changes to the crate's public API.